### ISSUE TITLE
Works not with language parameters

### ISSUE THREAD

**Issue by tanrak** [wilhelmer/mkdocs-localsearch#6]
I have tested the plugin and found that it does not work with other languages.
For example an excerpt from my mkdocs.yml:
```
theme:
    name: material
    custom_dir: theme
#    language: de

plugins:
    - search:
        lang:
#            - de
            - en
    - localsearch
```
Commented (with hash) it works, but commented out (without hash) it does not work.

Edit: Browser is Firefox

**Comment by wilhelmer** [comment]
This seems to be an issue with MkDocs Material.

With the configuration above, I get the following console output (FF, file:// protocol):

```console
TypeError: lunr.multiLanguage is undefined index.ts:148:15
Loading failed for the <script> with source “file:///C:/Temp/mkdocs-test/test/lunr/min/lunr.de.min.js”. index.html:1:1
Loading failed for the <script> with source “file:///C:/Temp/mkdocs-test/test/lunr/min/lunr.multi.min.js”. index.html:1:1
Loading failed for the <script> with source “file:///C:/Temp/mkdocs-test/test/lunr/min/lunr.stemmer.support.min.js”. index.html:1:1
```

With `search.lang` set to `de` only, I get:

```console
TypeError: e is undefined lunr.js:2678:2
Loading failed for the <script> with source “file:///C:/Temp/mkdocs-test/test/lunr/min/lunr.de.min.js”. index.html:1:1
Loading failed for the <script> with source “file:///C:/Temp/mkdocs-test/test/lunr/min/lunr.stemmer.support.min.js”. index.html:1:1
```

The issue seems to be related to both the `theme.language` and the `search.lang` settings.

If I use a web server, everything works as expected.

@squidfunk Could you please look into this? I have attached a sample project: [localsearch-issue-6.zip](https://github.com/wilhelmer/mkdocs-localsearch/files/4551043/localsearch-issue-6.zip)

**Comment by squidfunk** [comment]
Might be related to `iframe-worker` and `importScripts`, but needs further investigation. Currently, tight on time. Help appreciated.

**Comment by squidfunk** [comment]
So, I took some minutes to investigate. It's indeed related to `importScripts`, as they're loaded directly before setting up the index, which means that the stemmers haven't loaded yet. This is a limitation of the polyfill which is stated in the caveats section.

What I just tested is whether we can return a `Promise` from `importScripts`, and it seems to work. The latest master of `iframe-worker` contains the fix. The cool thing is that we could now use `async/await` in Material for MkDocs to `await` on `importScripts`, as awaiting a function that does not return a `Promise` is a no-op. However, I don't have the time to make the necessary adjustments in the worker code and test whether the non-local (i.e. default) case still works. If somebody could pick up from here, I'd be grateful. The code in question is [here](https://github.com/squidfunk/mkdocs-material/blob/master/src/assets/javascripts/integrations/search/worker/main/index.ts).

I attached the patched `iframe-worker` runtime which passes all tests: [polyfill.zip](https://github.com/wilhelmer/mkdocs-localsearch/files/4553306/polyfill.zip)

**Comment by wilhelmer** [comment]
Not sure if that's what you wanted me to do, but I downloaded your ZIP file, added `index.js` to `extra_javascript` in the sample project, built the project and tested using Material 5.1.4:

- FF & local: works ✔️ 
- FF & non-local: works ✔️
- Chrome & local: Search bar is [gracefully hidden](https://github.com/squidfunk/mkdocs-material/issues/1520) 🔴 
- Chrome & non-local: works ✔️

**Comment by squidfunk** [comment]
No, the search worker code needs to be adjusted to `await` on `importScripts`

**Comment by wilhelmer** [comment]
I tried, but couldn't get it to work. Sorry 😞 I have no experience with Typescript or `async/await`. I'm not a programmer, just a poor script kiddie.

**Comment by squidfunk** [comment]
Support for `await`able `importScripts` was just added to Material for MkDocs Insiders ☺️ This _should_ resolve the problems described in this issue, as the search index will now wait until the scripts have been loaded.

**Comment by wilhelmer** [comment]
Great, thanks. I'll check it out as soon as I have access to Insiders via my corporate account.

**Comment by squidfunk** [comment]
Great! We can work on this together from then on, so search works for your product after all. The funding goal for general availability is also not that far away.

**Comment by squidfunk** [comment]
The funding goal was hit, so let's revisit this issue when Material 6 is released, which _should_ fix this problem. I'm pretty sure it's going to arrive this weekend. If the issue persists, we can work on it.

**Comment by wilhelmer** [comment]
Uuuh v6, it's gonna be a major release. Yes, let's do that.

**Comment by squidfunk** [comment]
https://github.com/squidfunk/mkdocs-material/commit/9fe00ec0af47f9a3ee0c405d8692611ae4138269 fixes this issue, by detecting `iframe-worker` inside the search worker, fixing up the base URL and loading the stemmers via awaited `importScripts`. Note that two bugfix releases of `iframe-worker` followed, i.e. 0.1.8 and 0.1.9, so make sure to use the latest version, which is 0.1.9. My testing shows that this issue is now resolved. I tested with the following configuration.

``` yaml
theme:
  name: material
  language: de

plugins:
  - search
  - localsearch
```

~~The fixes are not yet released, so for now you'll have to check `master`. A release should follow this week with some more post v6 bugfixes 😊~~

Just released it as 6.0.1.
